### PR TITLE
NIFI-9959 Add UI Support for Sensitive Dynamic Properties

### PR DESCRIPTION
**Pull request by exceptionfactory**
# Summary

[NIFI-9959](https://issues.apache.org/jira/browse/NIFI-9959) Adds web user interface support for Sensitive Dynamic Properties in Processors, Controller Services, and Reporting Tasks based on framework support implemented in #6057 for NIFI-9958.

The implementation applies the `SupportsSensitiveDynamicProperties` annotation to `DBCPConnectionPool` and `ScriptedReportingTask` to enable evaluation of component types in addition to `InvokeHTTP`.

The UI implementation adds a new `Sensitive Value` field to the New Property dialog with radio button selections for `Yes` and `No`. The implementation follows the same approach as the `Sensitive Value` field in Parameter definitions. The internal processing collects sensitive dynamic property names for sending to the REST API on each invocation.

Components that do not support sensitive dynamic properties will show the `No` value selected, and the radio buttons will be disabled.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/propertytable/jquery.propertytable.js [hunk 1]
```diff
@@ -2028,17 +2045,26 @@
...
-                                    '<div class="setting-name">Property name</div>' +
+                                    '<div class="setting">' +
+                                        '<div class="setting-name">Property name</div>' +
                                         '<div class="setting-field new-property-name-container">' +
                                             '<input class="new-property-name" type="text"/>' +
                                         '</div>' +
                                     '</div>' +
+                                    '<div class="setting">' +
+                                        '<div class="setting-field new-property-sensitive-value-container">' +
+                                            '<div class="setting-name">Sensitive Value</div>' +
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/propertytable/jquery.propertytable.js:2056]: Can we add a tooltip icon here to provide guidance regarding this feature? Specifically, when it's supported and when the user is able to drive the sensitivity of the dynamic property.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/propertytable/jquery.propertytable.js [hunk 2]
```diff
@@ -2115,32 +2143,36 @@
...
+                                                previousValue: null,
+                                                value: null
+                                            }));
+
+                                            // select the new properties row
+                                            var row = propertyData.getRowById(existingItem.id);
+                                            propertyGrid.setActiveCell(row, propertyGrid.getColumnIndex('value'));
+                                            propertyGrid.editActiveCell();
+                                            propertyGrid.invalidateRow(row);
+                                            propertyGrid.render();
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/propertytable/jquery.propertytable.js:2160]: I believe calls to `invalidateRow` and `render` are preventing the `editActiveCell` from opening the editor for this row.

Comment (mcgilman) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/propertytable/jquery.propertytable.js:2160]: I assume the call to `invalidateRow` and `render` might have been needed for the updated descriptor to take effect. I didn't try but maybe we can reorder the calls here.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/propertytable/jquery.propertytable.js (hunk 1 above)
Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/propertytable/jquery.propertytable.js:2056]: Thanks for the suggestion, will add the icon with an explanation.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/propertytable/jquery.propertytable.js (hunk 2 above)
Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/propertytable/jquery.propertytable.js:2160]: Thanks for catching that issue, invalidating the row is necessary for the value formatter to evaluate the updated sensitive status, reordering the calls works as expected.

**Comment by exceptionfactory** [comment]
Thanks for the feedback @mcgilman! I pushed an update to incorporate the changes.

**Review by mcgilman — APPROVED** [review]
Thanks @exceptionfactory! Looks great. Will merge shortly once CI completes.

**Pull request merged**